File: market-application/src/main/java/com/example/market/application/command/PlaceBidCommand.java
package com.example.market.application.command;

import com.example.market.domain.catalog.SkuId;
import com.example.market.domain.shared.Money;
import com.example.market.domain.shared.UserId;

public record PlaceBidCommand(
        String idempotencyKey,
        UserId buyerId,
        SkuId skuId,
        Money bidPrice
) {}

File: market-application/src/main/kotlin/com/example/market/application/command/PlaceBidCommand.kt
package com.example.market.application.command

import com.example.market.domain.catalog.SkuId
import com.example.market.domain.shared.Money
import com.example.market.domain.shared.UserId

@JvmRecord
data class PlaceBidCommand(
    val idempotencyKey: String,
    val buyerId: UserId,
    val skuId: SkuId,
    val bidPrice: Money,
)
